tck/src/main/java/org/eclipse/microprofile/concurrency/tck/TckTest.java

```diff
@@ -30,14 +42,94 @@
 
     @Deployment
     public static WebArchive createDeployment() {
+        // build a JAR that provides fake 'ThreadPriority' context type
+        JavaArchive threadPriorityContextProvider = ShrinkWrap.create(JavaArchive.class, "threadPriorityContext.jar")
+                .addPackage(ThreadPriorityContextProvider.class.getPackage().getName())
+                .addAsServiceProvider(ThreadContextProvider.class, ThreadPriorityContextProvider.class);
+
+        // build a JAR that provides two fake context types: 'Buffer' and 'Label'
+        JavaArchive multiContextProvider = ShrinkWrap.create(JavaArchive.class, "bufferAndLabelContext.jar")
+                .addPackage(Buffer.class.getPackage().getName())
+                .addPackage(BufferContextProvider.class.getPackage().getName())
+                .addPackage(Label.class.getPackage().getName())
+                .addPackage(LabelContextProvider.class.getPackage().getName())
+                .addAsServiceProvider(ThreadContextProvider.class, BufferContextProvider.class, LabelContextProvider.class);
+
         return ShrinkWrap.create(WebArchive.class, TckTest.class.getSimpleName() + ".war")
-        .addClass(TckTest.class);
+                .addClass(TckTest.class)
+                .addAsLibraries(threadPriorityContextProvider, multiContextProvider);
+    }
+
+    @Test
+    public void builderForThreadContextIsProvided() {
+        Assert.assertNotNull(ThreadContext.builder(),
+                "MicroProfile Concurrency implementation does not provide a ThreadContext builder");
     }
 
     @Test
     public void providerSet() {
         ConcurrencyProvider provider = ConcurrencyProvider.instance();
-        Assert.assertNotNull(provider, "Provider is set");
+        Assert.assertNotNull(provider, "ConcurrencyProvider is not set");
     }
 
+    /**
+     * Verify that the MicroProfile Concurrency implementation finds third-party thread context providers
+     * that are made available to the ServiceLoader, allows their configuration via the ThreadContext builder,
+     * and correctly captures & propagates or clears these thread context types per the builder configuration.
+     * Subsequently verify that the MicroProfile Concurrency implementation properly restores thread context
+     * after the contextual action completes.
+     */
+    @Test
+    public void thirdPartyContextProvidersAreIncludedInThreadContext() {
```

Comment:
We should start splitting up TCK tests into different java classes, any categorization you come up with is fine by me, but we should have all the TCK tests for the spec in a single class.

Reply:
I suppose it makes sense to do that from the start. I'll add a commit that splits it out.